autopilot/contracts.go

```diff
@@ -39,3 +77,244 @@ func (ap *Autopilot) defaultContracts() ([]worker.Contract, error) {
 	}
 	return contracts, nil
 }
+
+func (ap *Autopilot) contractLoop() {
+	for {
+		select {
+		// TODO: use build var for loop interval
+		// TODO: add wakeChan (to trigger loop)
+		// TODO: run loop immediately
+		// TODO: logging
+		case <-ap.stopChan:
+			return
+		case <-time.After(time.Hour):
+		}
+
+		// don't run the contract loop if we are not synced
+		if !ap.store.Synced() {
+			continue
+		}
+
+		// fetch all contracts
+		cs, err := ap.defaultContracts()
+		if err != nil {
+			logErr(err)
+			continue
+		}
+
+		// run checks on the contract set
+		// TODO: dedupe existing contracts (?)
+		// TODO: check for IP range violations (?)
+
+		// run checks on the contracts individually
+		// TODO: separate loop (?)
+		csMap := make(map[string]int)
+		for i, c := range cs {
+			csMap[c.ID.String()] = i
+			// TODO: is host in host DB
+			// TODO: is max revision
+			// TODO: is offline
+			// TODO: is gouging
+			// TODO: is low score
+			// TODO: is out of funds
+			// TODO: update some contract metadata entity to reflect these checks
+		}
```

Comment:
Would factor some of the logic within this loop (including this) out into separate methods/functions.